src/features/ai-chat/utils/ws.ts

```
@@ -0,0 +1,202 @@
+    };
+
+    ws.onclose = (event) => {
+      console.log(`[WS] Disconnected from session ${sessionId}:`, {
+        code: event.code,
+        reason: event.reason,
+        wasClean: event.wasClean,
+        url: wsUrl
+      });
+      this.connections.delete(sessionId);
+      
+      const hasHandlers = this.handlers.get(sessionId)?.size || 0 > 0;
```

Biome (2.1.2)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)